samples/bluetooth/st_ble_sensor/src/main.c

```diff
@@ -0,0 +1,185 @@
+	0x41, 0x45, 0x22, 0x8e, 0x42, 0xfe, 0x00, 0x00);
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+#define ADV_LEN 12
+
+/* Advertising data */
+static u8_t manuf_data[ADV_LEN] = {
+	0x01 /*SKD version */,
+	0x83 /* STM32WB - P2P Server 1 */,
+	0x00 /* GROUP A Feature  */,
+	0x00 /* GROUP A Feature */,
+	0x00 /* GROUP B Feature */,
+	0x00 /* GROUP B Feature */,
+	0x00, /* BLE MAC start -MSB */
+	0x00,
+	0x00,
+	0x00,
+	0x00,
+	0x00, /* BLE MAC stop */
+};
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+	BT_DATA(BT_DATA_MANUFACTURER_DATA, manuf_data, ADV_LEN)
+};
+
+/* BLE connection */
+struct bt_conn *p_conn;
+/* Notification state */
+volatile u8_t notify_enable;
+
+static void mpu_ccc_cfg_changed(const struct bt_gatt_attr *attr, u16_t value)
+{
+	ARG_UNUSED(attr);
+	notify_enable = (value == BT_GATT_CCC_NOTIFY);
+	LOG_INF("Notification %s", notify_enable ? "enabled" : "disabled");
+}
+
+/* The STM32WB55xx board is acting as GATT server.
+ * The ST BLE Android app is the BLE GATT client.
+ */
+
+/* ST BLE Sensor GATT services and characteristic */
+
+BT_GATT_SERVICE_DEFINE(stsensor_svc,
+BT_GATT_PRIMARY_SERVICE(&st_service_uuid),
+BT_GATT_CHARACTERISTIC(&led_char_uuid.uuid,
+	BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE_WITHOUT_RESP,
+	BT_GATT_PERM_WRITE, NULL, recv, (void *)1),
+BT_GATT_CHARACTERISTIC(&but_notif_uuid.uuid, BT_GATT_CHRC_NOTIFY,
+	BT_GATT_PERM_READ, NULL, NULL, &but_val),
+BT_GATT_CCC(mpu_ccc_cfg_changed,
+	BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+);
+
+static ssize_t recv(struct bt_conn *conn,
+		    const struct bt_gatt_attr *attr, const void *buf,
+		    u16_t len, u16_t offset, u8_t flags)
+{
+	if (led_dev) {
+		if (led_state == true) {
+			led_on_off(0);
+			LOG_INF("Turn off LED");
+		} else {
+			led_on_off(1);
+			LOG_INF("Turn on LED");
+		}
+		led_state = !led_state;
+	}
+	return 0;
+}
+
+static void bt_ready(int err)
+{
+	if (err) {
+		LOG_ERR("Bluetooth init failed (err %d)", err);
+		return;
+	}
+	LOG_INF("Bluetooth initialized");
+
+	/* Start advertising */
+	err = bt_le_adv_start(BT_LE_ADV_CONN, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		LOG_ERR("Advertising failed to start (err %d)", err);
+		return;
+	}
+	p_conn = NULL;
+	LOG_INF("Configuration mode: waiting connections...");
+}
+
+static void connected(struct bt_conn *conn, u8_t err)
+{
+	if (err) {
+		LOG_ERR("Connection failed (err %u)", err);
+	} else {
+		LOG_INF("Connected");
+		p_conn = conn;
+	}
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	p_conn = NULL;
+	LOG_INF("Disconnected (reason %u)", reason);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+void main(void)
+{
+	int err;
+
+	err = button_init();
+	if (err)
+		LOG_ERR("Button init err");
```

Comment:
Missing { }

Also, log the error using %d

Reply:
ok